### DISCUSSION TITLE
Seasonal backgrounds unload when idle state is activated

### DISCUSSION THREAD

**Discussion by dekrain** [ppy/osu#16180] [Q&A] [answered]
After waiting 5 minutes idle in main menu, the game calls `SessionStatics.ResetValues()`, which resets currently loaded seasonal backgrounds, falling back to default backgrounds instead. The backgrounds will not reload until user reconnects to the server (for example by manually disconnecting and reconnecting).

A possible fix is to check whether seasonal backgrounds are already loaded
```diff
--- a/osu.Game/Configuration/SessionStatics.cs
+++ b/osu.Game/Configuration/SessionStatics.cs
@@ -21,7 +21,8 @@
public void ResetValues()
{
    ensureDefault(SetDefault(Static.LoginOverlayDisplayed, false));
    ensureDefault(SetDefault(Static.MutedAudioNotificationShownOnce, false));
    ensureDefault(SetDefault(Static.LowBatteryNotificationShownOnce, false));
    ensureDefault(SetDefault(Static.LastHoverSoundPlaybackTime, (double?)null));
-   ensureDefault(SetDefault<APISeasonalBackgrounds>(Static.SeasonalBackgrounds, null));
+   if (GetOriginalBindable<APISeasonalBackgrounds>(Static.SeasonalBackgrounds) == null)
+       ensureDefault(SetDefault<APISeasonalBackgrounds>(Static.SeasonalBackgrounds, null));
}
```

**Comment by bdach** [comment, marked as answer]
A better idea would be to exclude the "seasonal backgrounds" static from being reset at all by splitting the `InitialiseDefaults()` path from the session idle tracker path, as they serve different purposes.

**Reply by dekrain** [reply]
I can add a method like `IdleRevertDefaults` and call that instead of `ResetValues`.

**Reply by bdach** [reply]
Fine to use existing `InitialiseDefaults()` and `ResetValues()` methods imo, just with split implementations and appropriate xmldoc for the latter as to when it is used.

**Reply by dekrain** [reply]
Ok, I'll submit a PR
